Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs

```diff
@@ -565,8 +575,16 @@ public void ForceEndManipulation()
         /// </summary>
         public Vector3 GetPointerGrabPoint(uint pointerId)
         {
-            Assert.IsTrue(pointerIdToPointerMap.ContainsKey(pointerId));
-            return pointerIdToPointerMap[pointerId].GrabPoint;
+            int pointerDataListCount = pointerDataList.Count;
+            for (int i = 0; i < pointerDataListCount; i++)
```

Comment:
Maybe all of these for loops can be consolidated to a helper function rather than many for loops.

Reply:
Consolidated! These three, at least. The rest didn't follow a reasonably similar enough pattern.